spec/project-spec.coffee

```diff
@@ -197,6 +199,34 @@ describe "Project", ->
           atom.project.bufferForPath("b").then (anotherBuffer) ->
             expect(anotherBuffer).not.toBe buffer
 
+  describe ".repositoryForDirectory(directory)", ->
+    it "resolves to null when the directory does not have a repository", ->
+      theResult = "dummy_value"
```

Comment:
You could remove some boilerplate from these tests by putting all of the code inside a single `waitsForPromise` call:

```coffee
waitsForPromise ->
  atom.project.repositoryForDirectory(directory).then (result) ->
    # other assertions...
    expect(result).toBeNull()
```

Though the indentation would be one level greater, you could avoid the extra `theResult` variable and the `runs`, so I think it would be simpler to read.